sys-auth/sssd: remove unused file and add revision for updated init script

mm1ke opened the pull request:
Hi, 
This is another PR for sssd. In this PR i've added a new revision with a patch for the init script:
This patch fixes following:
* bug #499578: remove netmount dependency from the init script. As already mentioned in the bug, with netmount being a init dependency, someone can't login on a system when it's offline (sssd can cache credentials). I encountered this on my systems too and always had to "fix" this issue manually
* use openrc-run instead of runscript

This PR also removes a unused file from sssd. For the history i haven't merge that commit into the other one.

If the changes are welcomed i would also try to bring bring them upstream.

BTW, a version bump (bug #619444) would be nice as well :)

SoapGentoo commented on sys-auth/sssd/files/sssd-1.13.1-fix-init.patch:
```diff
@@ -0,0 +1,12 @@
+--- sssd-1.13.1-org/src/sysv/gentoo/sssd.in	2015-09-30 12:04:16.000000000 +0200
++++ sssd-1.13.1/src/sysv/gentoo/sssd.in	2017-08-06 17:31:30.692971705 +0200
```
please try to git imitate patches, i.e. make the header:
```
--- a/src/sysv/gentoo/sssd.in
+++ b/src/sysv/gentoo/sssd.in
```

1. Least number of bytes and still -p1
2. Version agnostic
3. Dates add no value

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,211 @@
+
+multilib_src_test() {
+	default
+}
```
check whether this is really necessary

mm1ke replied:
Sorry, but how can i check this correctly?
Is it enough to run tests with and without it and see if it behave the same way?

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,211 @@
+		--with-unicode-lib="glib2"
+		--disable-rpath
+		--disable-silent-rules
+		--sbindir=/usr/sbin
```
`--sbindir="${EPREFIX}"/usr/sbin`

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,211 @@
+		$(multilib_native_use_with selinux semanage)
+		$(use_enable locator krb5-locator-plugin)
+		$(multilib_native_use_with nfsv4 nfsv4-idmapd-plugin)
+		$(use_enable nls )
```
stray whitespace

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,211 @@
+		--with-crypto="libcrypto"
+		--with-initscript="sysv"
+
+		KRB5_CONFIG=/usr/bin/${CHOST}-krb5-config
```
`KRB5_CONFIG="${EPREFIX}"/usr/bin/${CHOST}-krb5-config`

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,211 @@
+		)
+
+		use locator || myconf+=(
+			KRB5_CONFIG=/bin/true
```
ditto

mm1ke commented:
Hi,

I've removed now the multilib_src_test() call as the package build's (with FEATURES="test") both ways normally. Guess it's not needed, but please let me know if there is a better way to check :)
I've also updated the metadata.xml file as repoman complained about wrong intentation

Please review

mm1ke commented:
Well, i've tried to test the "test" flag with abi_x86_32 enabled. Either way tests are failing now. Tests failing with: `src/util/sss_utf8.c:29:18: fatal error: glib.h: No such file or directory`
Glib is build with abi_x86_32 of course..

mm1ke commented:
Ok, i've fixed now multilib_src_test. Just `export GLIB_CFLAGS="-I${EPREFIX}/usr/include/glib-2.0` if tests run for the 32bit build. Hope this is correct this way.
Please review.

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,213 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
```
why no EAPI 6?

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,213 @@
+
+PYTHON_COMPAT=( python{2_7,3_4,3_5,3_6} )
+
+inherit eutils multilib pam linux-info autotools multilib-minimal python-r1 systemd toolchain-funcs
```
if you do EAPI 6, a lot of this can go away (eutils, multilib)

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,213 @@
+
+	eautoreconf
+
+	multilib_copy_sources
```
please try to avoid this and fix the build system properly

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,213 @@
+
+src_prepare() {
+	# bug 499578
+	epatch "${FILESDIR}"/${P}-fix-init.patch
```
stick this into global `PATCHES` before `pkg_setup` and instead of `epatch` + `epatch_user` call ust default

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,213 @@
+}
+
+src_configure() {
+	local native_dbus_cflags=$($(tc-getPKG_CONFIG) --cflags dbus-1)
```
don't override `src_configure`, rather use `multilib_is_native_abi`

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,213 @@
+	fi
+	if [[ "${PYTHON_TARGETS}" == *python3* ]]; then
+		myconf+=($(multilib_native_use_with python python3-bindings))
+	fi
```
*way* too complex:
```
if python_is_python3; then
	myconf+=($(multilib_native_use_with python python2-bindings))
else
	myconf+=($(multilib_native_use_with python python2-bindings))
fi
```

Anyhow, you are aware that you are using `python-r1`? i.e. you have to install it for _every_ python impl in `PYTHON_TARGETS`. I currently see you only doing it for one impl ever.

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,213 @@
+		emake -j1 DESTDIR="${D}" "${_at_args[@]}" install
+	else
+		# easier than playing with automake...
+		dopammod .libs/pam_sss.so
```
please fix this if possible

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,213 @@
+
+multilib_src_install_all() {
+	einstalldocs
+	prune_libtool_files --all
```
at the bottom of https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Handling_Libtool_Archives ...

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,213 @@
+
+	insinto /etc/sssd
+	insopts -m600
+	doins "${S}"/src/examples/sssd-example.conf
```
remove `"${S}"/`

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,213 @@
+
+	insinto /etc/logrotate.d
+	insopts -m644
+	newins "${S}"/src/examples/logrotate sssd
```
ditto

SoapGentoo commented on sys-auth/sssd/sssd-1.13.1-r1.ebuild:
```diff
@@ -0,0 +1,213 @@
+
+multilib_src_test() {
+	if ! multilib_is_native_abi; then
+		export GLIB_CFLAGS="-I${EPREFIX}/usr/include/glib-2.0"
```
1. `src_test` comes before `src_install`
2. Rather fix this in `configure.ac`
3. If 2. doesn't work, do it conditionally in `multilib_src_configure`: `use test && export GLIB_CFLAGS="-I${EPREFIX}/usr/include/glib-2.0"` (still slightly sucky though)

SoapGentoo commented:
@mm1ke please do the commits and do clean rebase

flappyports commented:
no activity
